source/fundamentals/connection.txt

```diff
@@ -0,0 +1,510 @@
+
+      <subsystem xmlns="urn:jboss:domain:naming:2.0">
+         <bindings>
+            <object-factory name="java:global/MyMongoClient" module="org.mongodb" class="com.mongodb.client.MongoClientFactory">
+               <environment>
+                  <property name="connectionString" value="mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"/>
+               </environment>
+            </object-factory>
+         </bindings>
+         <remote-naming/>
+      </subsystem>
+
+   .. note::
+
+      Replace the placeholder connection value URI here with a value that
+      points to your MongoDB installation.
+
+This will make a MongoClient instance accessible via the JNDI name
+``MyMongoClient`` in the ``java:global`` context.
+
+.. _jndi-tomcat:
+
+Tomcat
+~~~~~~
+
+1. Copy the ``mongo-java-driver.jar`` jar file into the ``lib`` directory
+   of your `Tomcat <http://tomcat.apache.org/>`__ installation.
+
+#. In ``context.xml`` of your application, add a resource that references
+   the ``MongoClientFactory`` class and the :ref:`connection string
+   <connection-uri>` for the MongoDB cluster:
+
+   .. code-block:: xml
+
+      <Resource name="mongodb/MyMongoClient"
+         auth="Container"
+         type="com.mongodb.MongoClient"
+         closeMethod="close"
+         factory="com.mongodb.client.MongoClientFactory"
+         singleton="true"
+         connectionString="mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"/>
+
+   .. note::
+
+      Replace the placeholder connection value URI here with a value that
+      points to your MongoDB installation.
+
+#. In ``web.xml`` of your application, add a reference to the
+   ``MongoClientFactory`` resource defined in the previous step:
+
+   .. code-block:: xml
+
+      <resource-ref>
+        <res-ref-name>
+          mongodb/MyMongoClient
+        </res-ref-name>
+        <res-type>
+          com.mongodb.MongoClient
+        </res-type>
+        <res-auth>
+          Container
+        </res-auth>
+      </resource-ref>
+
+This will make a ``MongoClient`` instance accessible via the JNDI name
+``mongodb/MyMongoClient`` in the ``java:comp/env`` context.
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname (or IP address)
+and port numbers of one or more members of the replica set separated by commas.
+Use the :manual:`replica set option
+</reference/connection-string/#urioption.replicaSet>` to specify the name
+of the replica set with which your application is connecting. When the
+replica set option is specified, the driver assumes that your
+connection string uses a seed list of multiple hosts instead of a single
+``host:port`` configuration.
```

Comment:
I like to describe examples immediately before presenting them. In this case, I might write:
"The connection string below specifies three hosts in the cluster and the replica set named 'myRs'"